cmd/builder/internal/builder/main.go

```diff
@@ -165,3 +223,43 @@ func processAndWrite(cfg Config, tmpl *template.Template, outFile string, tmplPa
 	defer out.Close()
 	return tmpl.Execute(out, tmplParams)
 }
+
+func (c *Config) coreModuleAndVersion() (string, string) {
+	module := "go.opentelemetry.io/collector"
+	if c.Distribution.RequireOtelColModule {
+		module += "/otelcol"
+	}
+	return module, "v" + c.Distribution.OtelColVersion
+}
+
+func (c *Config) allComponents() []Module {
+	// TODO: Use slices.Concat when we drop support for Go 1.21
+	return append(c.Exporters,
+		append(c.Receivers,
+			append(c.Processors,
+				append(c.Extensions,
+					c.Connectors...)...)...)...)
+}
+
+func (c *Config) readGoModFile() (string, map[string]string, error) {
```

Comment:
I haven't been able to add coverage for this function because in the code path, if I would expect this to fail, the previous `go get` [command](https://github.com/open-telemetry/opentelemetry-collector/pull/9897/files#diff-117127e4cf9f6587102dc92a7240d1e2c00944ee47ce23e02731f3a41466bdb1R147) fails first.

I can add a separate test that targets this function directly, otherwise not sure how to test this.